src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala

```diff
@@ -1951,6 +1980,66 @@ class AppsControllerTest extends UnitTest with GroupCreation with ScalatestRoute
       search(cmd = Some(""), id = Some(""), label = Some("")) should be(Set(app1, app2))
     }
 
+    new Fixture() {
+      Given("An unauthenticated request")
+      auth.authenticated = false
+      val app = """{"id":"/a/b/c","cmd":"foo","ports":[]}"""
+      val entity = HttpEntity(app.getBytes("UTF-8")).withContentType(ContentTypes.`application/json`)
+      groupManager.rootGroup() returns createRootGroup()
+
+      When("we try to fetch the list of apps")
```

Comment:
Nit: There is no need for `when`. Checkout `unauthenticatedRoute`. It already defines a `when`.